waggle-dance-core/src/main/java/com/hotels/bdp/waggledance/server/security/ReadWriteCreateAccessControlHandler.java

```diff
@@ -29,7 +46,27 @@ public boolean hasCreatePermission() {
 
   @Override
   public void databaseCreatedNotification(String name) {
-    // nothing to notify
+    // Notify to update mapped databases
+    List<String> mappedDatabases = null;
+    if (metaStore.getMappedDatabases() != null) {
+      mappedDatabases = new ArrayList<>(metaStore.getMappedDatabases());
+      if (!mappedDatabases.contains(name)) {
+        mappedDatabases.add(name);
+      }
+    }
+
+    AbstractMetaStore newMetaStore;
```

Comment:
All this code should be inside the `if (metaStore.getMappedDatabases() != null) {` that will avoid unnecessary updates.

Reply:
Even if it's null, we still need to do `federationService.update(metaStore, newMetaStore);` because we want `mappingsByDatabaseName` to contain the new database.

Reply:
ah yes you are right

Reply:
I only found out because a test was failing 😄  _(thank you, team standards XD)_